### DISCUSSION TITLE
Is `transfer-encoding: chunked` needed by pitchfork?

### DISCUSSION THREAD

**Discussion by ioquatix** [Shopify/pitchfork#138] [General]
Considering that every response is `connection: close`, is there any reason to use `transfer-encoding: chunked`? Maybe the server can be simplified in this regard?

**Comment by casperisfine** [comment]
> is there any reason to use transfer-encoding: chunked? 

Yes, it allows to confirm the stream properly ended by sending a 0 size chunk to act as EOF. If that EOF isn't received, then you didn't  get the full response (e.g. worker crashed or timed out).

**Reply by ioquatix** [reply]
Ah, yeah, that makes sense.
